pkg/services/authz/rbac/service.go

```diff
@@ -375,6 +376,15 @@ func (s *Service) getCachedIdentityPermissions(ctx context.Context, ns types.Nam
 	ctx, span := s.tracer.Start(ctx, "authz_direct_db.service.getCachedIdentityPermissions")
 	defer span.End()
 
+	latestUpdate, err := s.store.GetLatestUpdate(ctx, ns, store.LatestUpdateQuery{OrgID: ns.OrgID})
+	if err != nil {
+		return nil, err
+	}
+	// Reset cache if permissions have been updated in the db
+	if latestUpdate.Updated.After(time.Now().Add(-s.settings.CacheTTL)) {
```

Comment:
I've got some timezone issue here - time from DB returned as local time, so this comparison is incorrect.